### PR TITLE
Don't bind null pointers to database queries

### LINKED ISSUE #193: database bind text error for each file on importing new photos
bildschirmfoto von 2017-11-02 19 29 35

### PR DESCRIPTION
**Pull request by davidmhewitt**
Fixes #193 

The SQLite library was throwing assertions about trying to bind null pointers to strings in database queries.

Should be able to test this by running from the console and importing a new photo. The current master will probably spam the console with errors.

### CODE REVIEW

File: src/photos/PhotoMetadata.vala [hunk 1]
```diff
@@ -868,8 +868,11 @@ public class PhotoMetadata : MediaMetadata {
         }
     }
 
-    public override string? get_comment () {
-        return get_string_interpreted ("Exif.Photo.UserComment", PrepareInputTextOptions.DEFAULT & ~PrepareInputTextOptions.STRIP_CRLF);
+    public override string get_comment () {
+        var comment = get_string_interpreted ("Exif.Photo.UserComment",
+                                              PrepareInputTextOptions.DEFAULT & ~PrepareInputTextOptions.STRIP_CRLF);
+
+        return comment != null ? comment : "";
```
Comment (jeremypw) [inline src/photos/PhotoMetadata.vala]: This can be written `return comment ?? "" `

**Comment by davidmhewitt** [comment]
@jeremypw Well, I learned something new today, thanks!

Seems like quite a nice operator to remember!

**Comment by davidmhewitt** [comment]
Please don't merge this yet, I think I've had a better idea on how to do this.

**Comment by davidmhewitt** [comment]
This is now ready for review. When the `bind_text` method was previously being passed null pointers and throwing errors, it was actually binding `NULL` to those values in the database, so it makes sense to do that properly (i.e. without throwing errors) rather than modifying the way the data is stored by storing empty string as that may have some impact on queries etc...

**Pull request merged**